pkg/disruption/backend/sampler/remote.go

```diff
@@ -145,29 +149,14 @@ func StartInClusterMonitors(ctx context.Context, config *rest.Config) error {
 func deleteTestBed(ctx context.Context, kubeClient *kubernetes.Clientset) error {
 	// Remove daemonsets first to avoid trailing false-positive disruption intervals
 	dsClient := kubeClient.AppsV1().DaemonSets(namespace)
-	err := dsClient.DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
+	propagationPolicy := metav1.DeletePropagationForeground
+	err := dsClient.DeleteCollection(ctx, metav1.DeleteOptions{PropagationPolicy: &propagationPolicy}, metav1.ListOptions{})
+	if err != nil {
 		return fmt.Errorf("error removing daemonsets in namespace %s: %v", namespace, err)
 	}
-
-	timeLimitedCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
-	defer cancel()
-
-	if _, watchErr := watchtools.UntilWithSync(timeLimitedCtx,
-		cache.NewListWatchFromClient(
-			kubeClient.AppsV1().RESTClient(), "daemonsets", namespace, fields.Everything()),
-		&appsv1.DaemonSet{},
-		nil,
-		func(event watch.Event) (bool, error) {
-			return event.Type == watch.Deleted, nil
-		},
-	); watchErr != nil {
-		return fmt.Errorf("daemonsets in namespace %s didn't get destroyed: %v", namespace, watchErr)
-	}
-
 	nsClient := kubeClient.CoreV1().Namespaces()
-	err = nsClient.Delete(ctx, namespace, metav1.DeleteOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
+	err = nsClient.Delete(ctx, namespace, metav1.DeleteOptions{PropagationPolicy: &propagationPolicy})
```

Comment:
do not attempt to use propagation policy to control this.  It makes deletes less reliable overall and has been fragile over the years.